api/v0/common.go

```diff
@@ -0,0 +1,74 @@
+package v0
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+type Error struct {
+	Message string
+}
+
+var serverError []byte
+
+func init() {
+	// Make sure there is always an error to return in case encoding fails
+	e := Error{
+		Message: http.StatusText(http.StatusInternalServerError),
+	}
+
+	eb, err := json.Marshal(&e)
+	if err != nil {
+		panic(err)
+	}
+	serverError = eb
+}
+
+func EncodeError(err error) []byte {
+	if err == nil {
+		return nil
+	}
+	errMsg := Error{
+		Message: err.Error(),
+	}
+	data, err := json.Marshal(&errMsg)
+	if err != nil {
+		return serverError
+	}
+	return data
+}
+
+func DecodeError(data []byte) error {
+	if len(data) == 0 {
+		return nil
+	}
+	var e Error
+	err := json.Unmarshal(data, &e)
+	if err != nil {
+		return fmt.Errorf("cannot decode error message: %s", err)
```

Comment:
I am curious why `EncodeError` would return `serverError` but not here?

Reply:
`EncodeErorr` is used to send the error through the wire. `DecodeError` is used internally.